utils/window/window_test.go

```diff
@@ -275,3 +277,65 @@ func TestMinCapacity(t *testing.T) {
 	require.True(ok)
 	require.Equal(3, oldest)
 }
+
+// Tests timed eviction of elements using [EvictEvery] method
+func TestEvictEvery(t *testing.T) {
+	require := require.New(t)
+
+	clock := &mockable.Clock{}
+	// Using sync clock for this test to avoid race conditions
+	// with the eviction goroutine
+	clock.Sync()
```

Comment:
I assume we don't want to add locks to the clock since it's used everywhere.
To pass the data race detection we might be able to use atomics to store `clock.faked` and `clock.time` if we changed clock.time to stare `int64` unix representation instead of `time.Time`

This is a hack since changing two values is not atomic but these are only ever set for testing purposes when mocking the clock and would replace some sleeps in this test with `clock.Set` although we would still need to sleep for at least `2 *testEvictFrequency` in between expected evictions.